### PR TITLE
Enabled search for dbSNP column

### PR DESCRIPTION
**Pull request by onursumer**
Fix cBioPortal/cbioportal#7500

Describe changes proposed in this pull request:
- remove MyVariantInfo cache, and simplify MyVariantInfo columns
- add dbSNP column for the default table in `react-mutation-mapper` 
- enable search for dbSNP column

## Checks
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!

### CODE REVIEW

File: packages/cbioportal-utils/src/annotation/VariantAnnotationUtils.ts [hunk 1]
```diff
@@ -1,66 +1,15 @@
-import { MyVariantInfo, VariantAnnotation } from 'genome-nexus-ts-api-client';
+import { VariantAnnotation } from 'genome-nexus-ts-api-client';
 
 import { Mutation } from '../model/Mutation';
 import {
     extractGenomicLocation,
     genomicLocationString,
 } from '../mutation/MutationUtils';
 
-export function getMyVariantInfoAnnotationsFromIndexedVariantAnnotations(indexedVariantAnnotations?: {
```
Comment (onursumer) [inline packages/cbioportal-utils/src/annotation/VariantAnnotationUtils.ts:9]: moved to `MyVariantInfoAnnotationUtils`

File: packages/react-mutation-mapper/src/component/clinvar/ClinVarId.tsx [hunk 2]
```diff
@@ -1,20 +1,9 @@
 import { DefaultTooltip } from 'cbioportal-frontend-commons';
+import { getClinVarId } from 'cbioportal-utils';
 import { MyVariantInfo } from 'genome-nexus-ts-api-client';
 import { observer } from 'mobx-react';
 import * as React from 'react';
 
-export function getClinVarId(myVariantInfo?: MyVariantInfo): string | null {
```
Comment (onursumer) [inline packages/react-mutation-mapper/src/component/clinvar/ClinVarId.tsx:6]: moved to `MyVariantInfoAnnotationUtils`

File: packages/react-mutation-mapper/src/index.tsx [hunk 3]
```diff
@@ -65,7 +70,7 @@ export {
 } from './component/filter/MutationStatusBadgeSelector';
 export {
     default as GnomadFrequency,
-    calculateAlleleFrequency as calculateGnomadAllelFrequency,
+    calculateAlleleFrequency as calculateGnomadAlleleFrequency,
```
Comment (onursumer) [inline packages/react-mutation-mapper/src/index.tsx:73]: there was a typo in the function name

File: src/pages/patientView/clinicalInformation/PatientViewPageStore.ts [hunk 4]
```diff
@@ -1555,15 +1582,6 @@ export class PatientViewPageStore {
         );
     }
 
-    @cached get genomeNexusMyVariantInfoCache() {
```
Comment (onursumer) [inline src/pages/patientView/clinicalInformation/PatientViewPageStore.ts:1558]: we don't lazy load my variant info data anymore

File: src/shared/components/mutationTable/column/DbsnpColumnFormatter.tsx [hunk 5]
```diff
@@ -1,169 +1,97 @@
...
+                    indexedVariantAnnotations={indexedVariantAnnotations}
+                    indexedMyVariantInfoAnnotations={
+                        indexedMyVariantInfoAnnotations
+                    }
+                />
             </div>
         );
     }
 
-    private static getGenomeNexusDataFromCache(
```
Comment (onursumer) [inline src/shared/components/mutationTable/column/DbsnpColumnFormatter.tsx:31]: most of the code here refactored into `packages/react-mutation-mapper/src/component/column/Dbsnp.tsx`

File: src/shared/components/mutationTable/column/GnomadColumnFormatter.tsx [hunk 6]
```diff
@@ -1,174 +1,81 @@
...
+            IMyVariantInfoIndex | undefined
+        >
     ) {
-        const genomeNexusCacheData = GnomadColumnFormatter.getGenomeNexusDataFromCache(
-            data,
-            genomeNexusMyVariantInfoCache
-        );
         return (
             <span data-test="gnomad-column" data-test2={data[0].sampleId}>
-                {GnomadColumnFormatter.getGnomadDataViz(genomeNexusCacheData)}
```
Comment (onursumer) [inline src/shared/components/mutationTable/column/GnomadColumnFormatter.tsx:33]: now directly using the `Gnomad` component form `react-mutation-mapper`

**Comment by onursumer** [comment]
Since this PR removes the MyVariant info cache, related columns (gnomad, clinvar, dbSNP) will perform poorly until we merge and deploy genome-nexus/genome-nexus/pull/413.

**Comment by onursumer** [comment]
Thanks for testing @jjgao! The problem with "rs" should be fixed now. It was a case mismatch error, comparing the search keyword with the wrong case of the id.

**Comment by onursumer** [comment]
> Thanks Onur! Looks great! One minor thing not related to this pr, the dbsnp sorting sometimes is not correct, because we sort by string not by number. Here is an example in
> ascending order

@leexgh Good catch! Let's fix this in another PR.

**Comment by leexgh** [comment]
@onursumer sounds good!

**Comment by onursumer** [comment]
failing tests are unrelated, merging.

**Pull request merged**